chat/chat.go

```diff
@@ -1,105 +1,36 @@
 package chat
 
 import (
-	"crypto/rand"
-	"crypto/sha512"
-	"errors"
+	"time"
 
-	aes "github.com/Bit-Nation/panthalassa/crypto/aes"
+	prekey "github.com/Bit-Nation/panthalassa/chat/prekey"
+	db "github.com/Bit-Nation/panthalassa/db"
 	keyManager "github.com/Bit-Nation/panthalassa/keyManager"
+	bpb "github.com/Bit-Nation/protobuffers"
 	x3dh "github.com/Bit-Nation/x3dh"
-	doubleratchet "github.com/tiabc/doubleratchet"
+	dr "github.com/tiabc/doubleratchet"
+	ed25519 "golang.org/x/crypto/ed25519"
 )
 
-const ProtocolName = "pangea-chat"
-
-type Chat struct {
-	doubleRachetKeyStore doubleratchet.KeysStorage
-	x3dh                 x3dh.X3dh
-	km                   *keyManager.KeyManager
-}
-
-type Config struct {
-	// url of the websocket to connect to
-	WSUrl string
-	// url of the http endpoint
-	HttpEndpoint string
-	// access token for http endpoint and websocket
-	AccessKey string
-}
-
-// create a new chat
-func New(chatIdentityKey x3dh.KeyPair, km *keyManager.KeyManager, dRKeyStore doubleratchet.KeysStorage) (Chat, error) {
-
-	c := x3dh.NewCurve25519(rand.Reader)
-
-	x := x3dh.New(&c, sha512.New(), ProtocolName, chatIdentityKey)
-
-	return Chat{
-		doubleRachetKeyStore: dRKeyStore,
-		x3dh:                 x,
-		km:                   km,
-	}, nil
-
-}
-
-// create a new pre key bundle
-func (c *Chat) NewPreKeyBundle() (PanthalassaPreKeyBundle, error) {
-
-	// @todo usually this should be kept for longer time. Need to change that later on.
-	signedPreKey, err := c.x3dh.NewKeyPair()
-	if err != nil {
-		return PanthalassaPreKeyBundle{}, nil
-	}
-
-	oneTimePreKey, err := c.x3dh.NewKeyPair()
-	if err != nil {
-		return PanthalassaPreKeyBundle{}, err
-	}
-
-	chatIdKey, err := c.km.ChatIdKeyPair()
-	if err != nil {
-		return PanthalassaPreKeyBundle{}, err
-	}
-
-	idPubKeyStr, err := c.km.IdentityPublicKey()
-
-	preKeyB := PanthalassaPreKeyBundle{
-		PublicPart: PreKeyBundlePublic{
-			BChatIdentityKey: chatIdKey.PublicKey,
-			BSignedPreKey:    signedPreKey.PublicKey,
-			BOneTimePreKey:   oneTimePreKey.PublicKey,
-			BIdentityKey:     idPubKeyStr,
-		},
-		PrivatePart: PreKeyBundlePrivate{
-			OneTimePreKey: oneTimePreKey.PrivateKey,
-			SignedPreKey:  signedPreKey.PrivateKey,
-		},
-	}
-
-	if err := preKeyB.PublicPart.Sign(*c.km); err != nil {
-		return PanthalassaPreKeyBundle{}, err
-	}
-
-	return preKeyB, nil
-
-}
+const (
+	SignedPreKeyValidTimeFrame = time.Second * 3600 * 24 * 60
```

Comment:
SignedPreKeyValidTimeFrame will likely be a relatively long period of time which can be simplified by using 
`SignedPreKeyValidTimeFrame = time.Hour * 24 * 60`

Reply:
Done